File: auth/src/views/meta.rs
use crate::{types, utils};

use actix_web::{get, web, HttpResponse, Responder};
use serde::Serialize;

#[derive(Debug, Serialize)]
struct MetaResponse {
    status: String,
    data: types::Meta,
}

#[get("/meta")]
pub async fn meta_route(data: web::Data<types::AppContext>) -> impl Responder {
    let context = data.get_ref().clone();
    let meta = utils::get_meta_from_factors(context.meta_factors);

    HttpResponse::Ok().json(MetaResponse {
        status: "ok".to_string(),
        data: meta,
    })
}

File: email/routes/meta.go
package routes

import (
	"github.com/autobar-dev/services/email/types"
	"github.com/autobar-dev/services/email/utils"

	"github.com/labstack/echo/v4"
)

type MetaRouteResponse struct {
	Status string      `json:"status"`
	Data   *types.Meta `json:"data"`
}

func MetaRoute(c echo.Context) error {
	rest_context := c.(*types.RestContext)
	app_context := *(*rest_context).AppContext

	meta := utils.GetMetaFromFactors(app_context.MetaFactors)

	return rest_context.JSON(200, &MetaRouteResponse{
		Status: "ok",
		Data:   meta,
	})
}
